Add function to flush Router Cache

profburke opened the pull request:
Proposed implementation for issue #30 .

tanner0101 commented on Sources/Routing/Router.swift:
```diff
@@ -26,6 +26,26 @@ public class Router {
+    /// Request and re-cached. I.E. this function does not prune
+    /// the Branch.
+    @discardableResult
+    public func flushCache(for request: Request) -> Responder? {
```
Would it make sense to add `flushCache()` as well to flush the entire cache at once?

profburke commented:
@tanner0101 I think you raise a good point about api parity. I've added a function that flushes the entire cache. But that raises three issues:

1. I don't like the mismatch in return type of the two functions I've added. Although, I think maybe we just live with it because I do think having `flushCache(for:)` return the existing responder is useful.

2. I don't see a way to test the new function without breaking the encapsulation around `_cache`. Since removing the responder from the cache doesn't actually stop the router from responding to a given request, there's no "external" effect to test for.

3. We could add the ability to "unregister" a path. That would certainly make it possible to solve the testing issue raised above, but I'm not sure how useful the feature is. I certainly don't need it for my current application and, without any good reason to the contrary, I'd tend to stick with YAGNI.

tanner0101 commented:
1: i agree. esp with discardable result.
3: we can always add that in the future.

thanks!
